File: PA35.java
//LeetCode 11. Container With Most Water
//Time Complexity : O(n) // n is number of containers 
//Space Complexity: O(1)
class Solution {
    public int maxArea(int[] height) {
        
        int i=0;
        int j=height.length-1;
        int area=0;
        int temp=0;
        while(i<j){
            temp= (Math.min(height[i],height[j]) * (j-i));
            area= Math.max(temp, area);
            if(height[i]<height[j]){
                i++;
            }
            else{
                j--;
            }         
        }        
        return area;
    }
}

File: PA_35.py
#LC 11 - Container With Most Water
#Time Complexity - O(n)
#Space Complexity - O(1)
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        if not height:
            return 0
        low = 0
        high = len(height)-1
        ans = 0
        while low < high:
            area = (high - low) * min(height[low],height[high])
            ans = max(area,ans)
            #print(low,high)
            if height[low] < height[high]:
                low = low + 1
            else:
                high = high - 1
        
        return ans